contracts/pool/BundleService.sol

```diff
@@ -233,6 +233,103 @@ contract BundleService is
         bundleManager.lock(bundleNftId);
     }
 
+    /// @inheritdoc IBundleService
+    function stake(
+        IInstance instance,
+        NftId bundleNftId, 
+        Amount amount
+    ) 
+        external 
+        virtual
+        // TODO: restricted() (once #462 is done)
+    {
+        IBundle.BundleInfo memory bundleInfo = instance.getInstanceReader().getBundleInfo(bundleNftId);
+        StateId bundleState = instance.getInstanceReader().getMetadata(bundleNftId.toKey32(BUNDLE())).state;
```

Comment:
add function getBundleState to InstanceReader instead of doing this inline here (see eg InstanceReader.getPolicyState)